stock_demand_estimate/wizards/stock_demand_estimate_wizard.py

```diff
@@ -0,0 +1,200 @@
+                res.append(line.estimate_id.id)
+            else:
+                data = self._prepare_estimate_data(line)
+                estimate = self.env['stock.demand.estimate'].create(
+                    data)
+                res.append(estimate.id)
+        res = {
+            'domain': [('id', 'in', res)],
+            'name': _('Stock Demand Estimates'),
+            'src_model': 'stock.demand.estimate.wizard',
+            'view_type': 'form',
+            'view_mode': 'tree',
+            'res_model': 'stock.demand.estimate',
+            'type': 'ir.actions.act_window'
+        }
+        return res
+
+
+class StockDemandEstimateSheetLine(models.TransientModel):
+    _name = 'stock.demand.estimate.sheet.line'
+    _description = 'Stock Demand Estimate Sheet Line'
+
+    estimate_id = fields.Many2one(comodel_name='stock.demand.estimate')
+    date_range_id = fields.Many2one(
+        comodel_name='date.range',
+        string='Period')
+    location_id = fields.Many2one(comodel_name='stock.location',
+                                  string="Stock Location")
+    product_id = fields.Many2one(comodel_name='product.product',
+                                 string='Product')
+    value_x = fields.Char(string='Period')
+    value_y = fields.Char(string='Product')
+    product_uom_qty = fields.Float(
+        string="Quantity", digits_compute=dp.get_precision('Product UoM'))
+
+
+class DemandEstimateWizard(models.TransientModel):
+    _name = 'stock.demand.estimate.wizard'
+    _description = 'Stock Demand Estimate Wizard'
+
+    date_start = fields.Date(string="Date From", required=True)
+    date_end = fields.Date(string="Date To", required=True)
+    date_range_type_id = fields.Many2one(string='Date Range Type',
+                                         comodel_name='date.range.type',
+                                         required=True)
+    location_id = fields.Many2one(comodel_name="stock.location",
+                                  string="Location", required=True)
+    product_ids = fields.Many2many(
+        comodel_name="product.product",
+        string="Products")
+
+    @api.multi
+    def _prepare_demand_estimate_sheet(self):
+        self.ensure_one()
+        return {
+            'date_start': self.date_start,
+            'date_end': self.date_end,
+            'date_range_type_id': self.date_range_type_id.id,
+            'location_id': self.location_id.id,
+        }
+
+    @api.multi
+    def create_sheet(self):
+        self.ensure_one()
+        if not self.product_ids:
+            raise UserError(_('You must select at lease one product.'))
+
+        context = {
+            'default_date_start': self.date_start,
+            'default_date_end': self.date_end,
+            'default_date_range_type_id': self.date_range_type_id.id,
+            'default_location_id': self.location_id.id,
+            'default_product_ids': self.product_ids.ids
```

Comment:
As _product_ids_ is not an attribute of the model _stock.demand.estimate.sheet_, it would be better to pass this parameter without the `default_` prefix, in order to avoid confusion.